packages/taquito-http-utils/src/taquito-http-utils.ts

```diff
@@ -74,60 +67,63 @@ export class HttpBackend {
   /**
    *
    * @param options contains options to be passed for the HTTP request (url, method and timeout)
-   * @throws {@link HttpRequestFailed} | {@link HttpResponseError}
+   * @throws {@link HttpRequestFailed} | {@link HttpResponseError} | {@link HttpTimeoutError}
    */
   async createRequest<T>(
-    { url, method, timeout = this.timeout, query, headers = {}, json = true }: HttpRequestOptions,
+    { url, method, query, timeout = this.timeout, headers = {}, json = true }: HttpRequestOptions,
     data?: object | string
   ) {
+    // Serializes query params
     const urlWithQuery = url + this.serialize(query);
-    let resType: ResponseType;
-    let transformResponse = undefined;
 
+    // Adds default header entry if there aren't any Content-Type header
     if (!headers['Content-Type']) {
       headers['Content-Type'] = 'application/json';
     }
 
-    if (!json) {
-      resType = ResponseType.TEXT;
-      transformResponse = [<Type>(v: Type) => v];
-    } else {
-      resType = ResponseType.JSON;
-    }
+    // Creates a new AbortController instance to handle timeouts
+    const controller = new AbortController();
```

Comment:
please correct if i am mistaken just don't know where is AbortController is defined or imported to this file

Reply:
read the node doc realizing it's native in node, will resolve this convo

Reply:
AbortController is built in from node